structure_tests/drivers/driver.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2017 Google Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package drivers
+
+import (
+	"os"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/runtimes-common/structure_tests/types/unversioned"
+)
+
+type Driver interface {
+	Info() string
+
+	Setup(t *testing.T, envVars []unversioned.EnvVar, fullCommand []string,
+		shellMode bool, checkOutput bool)
```

Comment:
I think we should probably drop shellMode as a feature, WDYT? It's pretty easy to convert for users:

[a, b, c] turns into ['sh', '-c', 'a b c']

Reply:
I think I'm fine with this. The only thing we lose is the ability to check the exit code of an intentionally malformed command: it will always return -1, even though the underlying process could have returned something else (e.g. 127 for a non-existent binary), but I don't think this is useful enough to jump through hoops to support